apps/nextjs/src/pages/auth/signin.tsx

```diff
@@ -18,15 +17,44 @@ export const getServerSideProps: GetServerSideProps = async (ctx) => {
   const fromOAuth =
     origin?.includes("google") ||
     origin?.includes("live") ||
-    origin?.includes("microsoft");
+    origin?.includes("microsoft") ||
+    origin?.includes("stripe");
 
-  // Redirect to the callbackUrl if we're coming from an OAuth provider
+  // This means we have completed the sign in flow
   if (fromOAuth && callbackUrl) {
+    // @ts-expect-error - Not sure how to type this to accept GetServerSidePropsContext across packages - but they are compatible
+    const trpcCtx = await createContext(ctx);
+    const caller = appRouter.createCaller(trpcCtx);
```

Comment:
So the `caller` lets us call trpc procedures server-side?

Reply:
No idea about the types either by the way

Reply:
Yes, `createCaller` is how you do server-side calls

To fix the type, the api package would need to add next as dependency since we'd need the `GetServerSidePropsContext` type - I think ignoring the error is fine for now.